### PR TITLE
Make a copy of the logfile before rollback

### PR DESCRIPTION
**Pull request by ludeeus**
Creates `home-assistant-rollback.log` from `home-assistant.log` before rollback, so the reason for rollback is not lost

### CODE REVIEW

File: supervisor/homeassistant.py [hunk 1]
```diff
@@ -359,6 +359,13 @@ async def _update(to_version: str) -> None:
         # Update going wrong, revert it
         if self.error_state and rollback:
             _LOGGER.critical("HomeAssistant update fails -> rollback!")
+            # Make a copy of the current log file if it exsist
+            logfile = Path(self.sys_config.path_homeassistant, "home-assistant.log")
+            if logfile.exists():
+                backup = Path(
+                    self.sys_config.path_homeassistant, "home-assistant-rollback.log"
+                )
+                shutil.copy(logfile.as_posix(), backup.as_posix())
```
Comment (balloob) [inline supervisor/homeassistant.py:368]: `shutil.copy` doesn't take `Path` objects?

Comment (ludeeus) [reply supervisor/homeassistant.py:368]: Looks like it does, I'll change it to use that.

File: supervisor/homeassistant.py [hunk 2]
```diff
@@ -359,6 +359,13 @@ async def _update(to_version: str) -> None:
         # Update going wrong, revert it
         if self.error_state and rollback:
             _LOGGER.critical("HomeAssistant update fails -> rollback!")
+            # Make a copy of the current log file if it exsist
+            logfile = Path(self.sys_config.path_homeassistant, "home-assistant.log")
```
Comment (balloob) [inline supervisor/homeassistant.py:363]: If `path_homeassistant` is a Path object, this works too:
```suggestion
            logfile = self.sys_config.path_homeassistant / "home-assistant.log"
```

File: supervisor/homeassistant.py [hunk 3]
```diff
@@ -359,6 +359,17 @@ async def _update(to_version: str) -> None:
...
             _LOGGER.critical("HomeAssistant update fails -> rollback!")
+            # Make a copy of the current log file if it exsist
+            logfile = self.sys_config.path_homeassistant / "home-assistant.log"
+            if logfile.exists():
+                backup = (
+                    self.sys_config.path_homeassistant / "home-assistant-rollback.log"
+                )
+
+                shutil.copy(logfile, backup)
+                _LOGGER.info(
```
Comment (balloob) [inline supervisor/homeassistant.py:370]: They will never see this, right?

Comment (ludeeus) [reply supervisor/homeassistant.py:370]: It will show in the supervisor log

Comment (ludeeus) [reply supervisor/homeassistant.py:370]:
```
20-08-03 10:40:12 CRITICAL (MainThread) [supervisor.homeassistant] HomeAssistant update fails -> rollback!
20-08-03 10:40:12 INFO (MainThread) [supervisor.homeassistant] A backup of the logfile is stored in /config/home-assistant-rollback.log
```

**Pull request merged**